packages/fx-core/src/plugins/resource/frontend/configs.ts

```diff
@@ -105,25 +102,15 @@ export class FrontendConfig {
   }
 
   static getSubscriptionId(ctx: PluginContext): string {
-    const result = isArmSupportEnabled()
-      ? getSubscriptionIdFromResourceId(FrontendConfig.getStorageResourceId(ctx))
-      : FrontendConfig.getConfig<string>(
-          DependentPluginInfo.SubscriptionId,
-          ctx.envInfo.state.get(DependentPluginInfo.SolutionPluginName)
-        );
+    const result = getSubscriptionIdFromResourceId(FrontendConfig.getStorageResourceId(ctx));
```

Comment:
I have an assumption for getting subscriptionId, ie, the priority would be solution > others.
If the assumption is true, better to get subsId from solution first, if missed, then get it from other places for fallback.
What do you say?

Reply:
I have different opinion. I think the resource Id exactly describes what the resource is and where it is (subscription and resource group), rather than solution's config.
Anyway, this PR is just to refactor. We can discuss it later.

Reply:
Agree - not so critical, you can choose for your preference.